mgit/lib/utils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const execSh = require( 'exec-sh' );
+
+const utils = {
+	/**
+	 * Executes a command.
+	 *
+	 * @param {String} command Command to execute.
+	 * @returns {Promise}
+	 */
+	exec( command ) {
+		return new Promise( ( resolve, reject ) => {
+			execSh( command, true, ( err, stdout, stderr ) => {
```

Comment:
Why async? Do we gain anything by using async execSh? I don't think so, because we've got a linear flow of each process (one command needs to finish before the next one is started). So it's better to use synchronous versions of IO and other methods, because then the code is easier to read.

I can see that `exec-sh` doesn't have a synchronous mode, so this means that perhaps `shelljs` will better fit our use case?